src/commands/slash/RuleCommand.ts

```diff
@@ -0,0 +1,43 @@
+import {AutocompleteInteraction, ColorResolvable, CommandInteraction, MessageEmbed} from "discord.js";
+import {Discord, Slash, SlashOption} from "discordx";
+import getConfigValue from "../../utils/getConfigValue";
+import GenericObject from "../../interfaces/GenericObject";
+
+interface Rule {
+	name: string;
+	triggers: string[];
+	description: string;
+}
+
+@Discord()
+class RuleCommand {
+	@Slash("rule")
+	async onInteract(
+		@SlashOption("rule", {
+			autocomplete: (interaction: AutocompleteInteraction) => {
+				const rule = getConfigValue<Rule[]>("rules").map(it => ({name: it.name, value: it.triggers[0]}));
+
+				interaction.respond(rule);
```

Comment:
I don't really like how this callback lives within the decorator. I wonder if it would be better to read the rules in separately and store them in a `const`?